Lyrical compatibility - time to source newer lely-core libraries?

Ran a basic installation test using the vanilla Docker image:
osrf/ros  || lyrical-desktop-full || sha256:fd6d22302e3b7cca28b9f1d636ff64f9408b5445251ac85c10dc958ba72df9da


        Starting >>> lely_core_libraries
        --- stderr: canopen_interfaces                                                                                 
        CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
          Compatibility with CMake < 3.10 will be removed from a future version of
          CMake.

          Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
          to tell CMake that the project requires at least <min> but has been updated
          to work with policies introduced by <max> or earlier.


        ---
        Finished <<< canopen_interfaces [9.53s]
        Starting >>> canopen_utils
        /usr/lib/python3/dist-packages/setuptools/_distutils/dist.py:289: UserWarning: Unknown distribution option: 'tests_require'
          warnings.warn(msg)
        --- stderr: canopen_utils                                                                         
        /usr/lib/python3/dist-packages/setuptools/_distutils/dist.py:289: UserWarning: Unknown distribution option: 'tests_require'
          warnings.warn(msg)
        ---
        Finished <<< canopen_utils [0.83s]
        --- stderr: lely_core_libraries                                
        CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
          Compatibility with CMake < 3.10 will be removed from a future version of
          CMake.

          Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
          to tell CMake that the project requires at least <min> but has been updated
          to work with policies introduced by <max> or earlier.


        Cloning into 'upstream'...
        /usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py:90: SetuptoolsDeprecationWarning: setup.py install is deprecated.
        !!

                ********************************************************************************
                Please avoid running ``setup.py`` directly.
                Instead, use pypa/build, pypa/installer or other
                standards-based tools.

                See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
                ********************************************************************************

        !!
          self.initialize_options()
        HEAD is now at fb735b79 nmt: remove EMCY messages when life-guarding/heartbeat errors are resolved
        Checking patch configure.ac...
        Hunk #1 succeeded at 577 (offset 8 lines).
        Checking patch python/dcf-tools/Makefile.am...
        Applied patch configure.ac cleanly.
        Applied patch python/dcf-tools/Makefile.am cleanly.
        configure.ac:17: warning: The macro 'AC_PROG_CC_STDC' is obsolete.
        configure.ac:17: You should run autoupdate.
        ./lib/autoconf/c.m4:1669: AC_PROG_CC_STDC is expanded from...
        configure.ac:17: the top level
        configure.ac:442: warning: AC_CHECK_HEADERS($ixxat_header): you should use literals
        ./lib/autoconf/headers.m4:217: AC_CHECK_HEADERS is expanded from...
        lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
        lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
        configure.ac:442: the top level
        configure.ac:17: installing './compile'
        configure.ac:6: installing './config.guess'
        configure.ac:6: installing './config.sub'
        configure.ac:7: installing './install-sh'
        configure.ac:7: installing './missing'
        configure.ac:610: installing './tap-driver.sh'
        src/can/Makefile.am: installing './depcomp'
        parallel-tests: installing './test-driver'
        In file included from /root/colcon_ws/build/lely_core_libraries/upstream/src/util/spscring.c:25:
        /root/colcon_ws/build/lely_core_libraries/upstream/src/util/spscring.c: In function ‘spscring_init’:
        /root/colcon_ws/build/lely_core_libraries/upstream/include/lely/util/spscring.h:102:25: error: implicit declaration of function ‘ATOMIC_VAR_INIT’; did you mean ‘ATOMIC_FLAG_INIT’? [-Wimplicit-function-declaration]
          102 |                         ATOMIC_VAR_INIT(0), \
              |                         ^~~~~~~~~~~~~~~
        /root/colcon_ws/build/lely_core_libraries/upstream/src/util/spscring.c:52:34: note: in expansion of macro ‘SPSCRING_INIT’
           52 |         *ring = (struct spscring)SPSCRING_INIT(size);
              |                                  ^~~~~~~~~~~~~
        At top level:
        cc1: note: unrecognized command-line option ‘-Wno-static-in-inline’ may have been intended to silence earlier diagnostics
        gmake[6]: *** [Makefile:668: liblely_util_la-spscring.lo] Error 1
        gmake[6]: *** Waiting for unfinished jobs....
        gmake[5]: *** [Makefile:425: all-recursive] Error 1
        gmake[4]: *** [Makefile:489: all-recursive] Error 1
        gmake[3]: *** [Makefile:417: all] Error 2
        gmake[2]: *** [CMakeFiles/upstr_lely_core_libraries.dir/build.make:86: upstr_lely_core_libraries-prefix/src/upstr_lely_core_libraries-stamp/upstr_lely_core_libraries-build] Error 2
        gmake[1]: *** [CMakeFiles/Makefile2:160: CMakeFiles/upstr_lely_core_libraries.dir/all] Error 2
        gmake: *** [Makefile:146: all] Error 2
        ---
        Failed   <<< lely_core_libraries [19.0s, exited with code 2]

        Summary: 2 packages finished [19.1s]
          1 package failed: lely_core_libraries

--- a/.github/workflows/rolling.yml
+++ b/.github/workflows/rolling.yml
@@ -21,6 +21,8 @@ jobs:
             ROS_REPO: testing
           - ROS_DISTRO: kilted
             ROS_REPO: testing
+          - ROS_DISTRO: lyrical
+            ROS_REPO: testing
     env:
       CCACHE_DIR: "${{ github.workspace }}/.ccache"
     steps:

--- a/.github/workflows/rolling_documentation.yml
+++ b/.github/workflows/rolling_documentation.yml
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, jazzy, kilted]
+        ROS_DISTRO: [rolling, jazzy, kilted, lyrical]
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:

--- a/canopen_core/ConfigExtras.cmake
+++ b/canopen_core/ConfigExtras.cmake
@@ -12,4 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-find_package(Boost REQUIRED system thread)
+find_package(Boost REQUIRED thread)

--- a/canopen_core/src/lifecycle_manager.cpp
+++ b/canopen_core/src/lifecycle_manager.cpp
@@ -80,8 +80,7 @@ bool LifecycleManager::load_from_config()
     auto node_id_opt = config_->get_entry<uint8_t>(device_name, "node_id");
     if (!node_id_opt.has_value())
     {
-      RCLCPP_WARN(
-        this->get_logger(), "Skipping %s: missing node_id entry.", device_name.c_str());
+      RCLCPP_WARN(this->get_logger(), "Skipping %s: missing node_id entry.", device_name.c_str());
       continue;
     }
     uint8_t node_id = node_id_opt.value();

--- a/lely_core_libraries/CMakeLists.txt
+++ b/lely_core_libraries/CMakeLists.txt
@@ -11,12 +11,13 @@ ExternalProject_Add(upstr_lely_core_libraries    # Name for custom target
   INSTALL_DIR "${CMAKE_INSTALL_PREFIX}"  # Installation prefix
   BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/build
   GIT_REPOSITORY https://gitlab.com/lely_industries/lely-core.git
-  GIT_TAG fb735b79cab5f0cdda45bc5087414d405ef8f3ab
+  GIT_TAG 9e3267d26018f6f6babd50786f6ae2af89cc57ea
   TIMEOUT 60
-  #UPDATE step apply patch to fix dcf-tools install
+  #UPDATE step apply patch to fix dcf-tools install and C23/GCC14 compat
   UPDATE_COMMAND
   COMMAND git reset --hard
   COMMAND git apply --whitespace=fix --reject ${CMAKE_CURRENT_SOURCE_DIR}/patches/0001-Fix-dcf-tools.patch
+  COMMAND git apply --whitespace=fix --reject ${CMAKE_CURRENT_SOURCE_DIR}/patches/0002-Fix-C23-compat.patch
   #CONFIGURE step execute autoreconf and configure
   CONFIGURE_COMMAND autoreconf -i <SOURCE_DIR>
   COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --disable-cython --disable-doc --disable-tests --disable-static --disable-diag

--- a/lely_core_libraries/patches/0002-Fix-C23-compat.patch
+++ b/lely_core_libraries/patches/0002-Fix-C23-compat.patch
@@ -1,0 +1,57 @@
+From 0000000000000000000000000000000000000002 Mon Sep 17 00:00:00 2001
+From: Build Fix <noreply@example.com>
+Date: Sat, 24 May 2026 00:00:00 +0000
+Subject: [PATCH] Fix C23/GCC14 build errors
+
+ATOMIC_VAR_INIT was deprecated in C17 and removed in C23; GCC 14 and
+Clang 17 no longer define it. Replace every occurrence in
+SPSCRING_INIT with a plain parenthesised initializer, which is valid
+for _Atomic variables since C11.
+
+char tab[64] initialised with a 65-character string literal (64 chars
++ NUL) causes -Werror=unterminated-string-initialization on GCC 14+.
+Widening the array to 65 silences the diagnostic without changing
+any runtime behaviour.
+---
+ include/lely/util/spscring.h | 8 ++++----
+ src/util/print.c             | 2 +-
+ 2 files changed, 5 insertions(+), 5 deletions(-)
+
+diff --git a/include/lely/util/spscring.h b/include/lely/util/spscring.h
+index 849b583f..a0c96e3f 100644
+--- a/include/lely/util/spscring.h
++++ b/include/lely/util/spscring.h
+@@ -99,14 +99,14 @@ struct spscring {
+ 		{ \
+ 			{ (size), 0, 0, 0 }, \
+ 			{ 0 }, \
+-			ATOMIC_VAR_INIT(0), \
++			(0), \
+ 			{ 0 }, \
+-			{ ATOMIC_VAR_INIT(0), NULL, NULL }, \
++			{ (0), NULL, NULL }, \
+ 			{ 0 }, \
+ 		}, \
+ 		{ \
+-			{ (size), 0, 0, 0 }, { 0 }, ATOMIC_VAR_INIT(0), { 0 }, \
+-					{ ATOMIC_VAR_INIT(0), NULL, NULL }, \
++			{ (size), 0, 0, 0 }, { 0 }, (0), { 0 }, \
++					{ (0), NULL, NULL }, \
+ 					{ 0 }, \
+ 		} \
+ 	}
+diff --git a/src/util/print.c b/src/util/print.c
+index 94d8aa59..035c7621 100644
+--- a/src/util/print.c
++++ b/src/util/print.c
+@@ -262,7 +262,7 @@ size_t
+ print_base64(char **pbegin, char *end, const void *ptr, size_t n)
+ {
+ 	// clang-format off
+-	static const char tab[64] =
++	static const char tab[65] =
+ 			"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdef"
+ 			"ghijklmnopqrstuvwxyz0123456789+/";
+ 	// clang-format on
+--
+2.34.1

